code/go/0chain.net/smartcontract/dbs/event/authorizer_aggregate.go

```diff
@@ -115,9 +115,10 @@ func (edb *EventDb) calculateAuthorizerAggregate(gs *globalSnapshot, round, limi
 	logging.Logger.Debug("getting ids", zap.Strings("ids", ids))
 
 	var currentAuthorizers []Authorizer
-	result := edb.Store.Get().
-		Raw("SELECT * FROM Authorizers WHERE id in (select id from authorizer_temp_ids ORDER BY ID limit ? offset ?)", limit, offset).
-		Scan(&currentAuthorizers)
+	result := edb.Store.Get().Model(&Miner{}).
+		Where("authorizers.id in (select id from authorizer_temp_ids ORDER BY ID limit ? offset ?)", limit, offset).
```

Comment:
why it is joined? you need to calculate it in aggregate

Reply:
It's calculated in aggregate [here](https://github.com/0chain/0chain/blob/b2e023590f2af1fe2fbf9e33299af919ae440bc3/code/go/0chain.net/smartcontract/dbs/event/provider.go#L48-L49) but it's calculated from `miners` and `miner_snapshots` table. Not joining it gives `miner.Reward.TotalReward = 0` because it's not brought from the DB. Also updating miner_snapshot depends on this query so also its `TotalReward` is always 0.

Reply:
Kindly see https://github.com/0chain/0chain/blob/b2e023590f2af1fe2fbf9e33299af919ae440bc3/code/go/0chain.net/smartcontract/dbs/event/miner_aggregate.go#L131

Reply:
why is it miner?